tests/lint_message_convention.rs

```diff
@@ -0,0 +1,110 @@
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+use regex::RegexSet;
+
+#[derive(Debug)]
+struct Message {
+    path: PathBuf,
+    bad_lines: Vec<String>,
+}
+
+impl Message {
+    fn new(path: PathBuf) -> Self {
+        let content: String = std::fs::read_to_string(&path).unwrap();
+        // we don't want the first letter after "error: ", "help: " ... to be capitalized
+        // also no puncutation (except for "?" ?) at the end of a line
+        let regex_set: RegexSet = RegexSet::new(&[
+            r"error: [A-Z]",
+            r"help: [A-Z]",
+            r"warning: [A-Z]",
+            r"note: [A-Z]",
+            r"try this: [A-Z]",
+            r"error: .*[.!]$",
+            r"help: .*[.!]$",
+            r"warning: .*[.!]$",
+            r"note: .*[.!]$",
+            r"try this: .*[.!]$",
+        ])
+        .unwrap();
+
+        // sometimes the first character is capitalized and it is legal (like in "Iterator...") or
+        // we want to ask a question ending in "?"
+        let exceptions_set: RegexSet = RegexSet::new(&[
+            r".*error: I see you're using a LinkedList! Perhaps you meant some other data structure?",
+            r".*C-like enum variant discriminant is not portable to 32-bit targets",
+            r".*Iterator::step_by(0) will panic at runtime",
+            r".*did you mean `unix`?",
+            r".*the arguments may be inverted...",
+            r".*Intel x86 assembly syntax used",
+            r".*AT&T x86 assembly syntax used",
+            r".*remove .*the return type...",
+            r"note: Clippy version: .*",
+            r"the compiler unexpectedly panicked. this is a bug.",
+            r".*help: I think you meant: .*",
+            r"Iterator.* will panic at runtime",
+        ])
```

Comment:
Can this be more generalized? For `Iterator::step_by(0)`, you could detect `Captialized::word`. For `did you mean unix`, you could detect the question mark. You can have a whitelist of proper noun first words like "I", "Intel" or "C-like". For multiple sentences, you can detect `. Capitalized..`.

Reply:
Hm. I would rather have a false positive (let's say "Iterator::count() foo bla") trigger a CI failure just to be immediately added to the allowlist than have a false negative slip through because allow-list regex is too general.

Reply:
Yeah, I'm also fine with having a hardcoded list. (With my two comments applying to 4 messsages above addressed).